internal/pkg/config/environment.go

```diff
@@ -0,0 +1,42 @@
+/*******************************************************************************
+ * Copyright 2019 Dell Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+
+package config
+
+import (
+	"os"
+	"strconv"
+)
+
+const (
+	envKeyPrefix = "edgex_registry_"
```

Comment:
- As far as interacting with the Registry is concerned, please note that the SMA will (in all likelihood) use the go-mod-registry PutConfigurationValue() since it will not have the services config struct the is passed to `PutConfiguration`. And since the SMA will go directly to the Registry, rather than the service, the need exists for knowing the proper path in Consul for the (given) key.
- The Registry Client is created for a specific service with that services service key. That takes care of the base path.
- So we need to know the (trailing) `sub-path` for the specific configuration value.
- Here, in the context of the logic encoded in `environment.go`, would this logic have responsibility for handling that aspect? (i.e. "figure out the (trailing) `sub-path` for the specific configuration value").

Reply:
The changes in this PR override the settings in a configuration.toml file.  They know nothing of the mapping from configuration.toml to registry.  The referenced source `internal/pkg/config/environment.go` simply host, port and type in a RegistryInfo struct that's contained within a service's ConfigurationStruct struct.

Reply:
That makes sense 👍 Thanks for confirming. I'm good with it.